RecoBTag/FeatureTools/plugins/DeepDoubleCvBTagInfoProducer.cc

```diff
@@ -0,0 +1,349 @@
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/Candidate/interface/Candidate.h"
+
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+
+#include "DataFormats/BTauReco/interface/BoostedDoubleSVTagInfo.h"
+
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+
+#include "DataFormats/BTauReco/interface/DeepDoubleCvBFeatures.h"
+#include "DataFormats/BTauReco/interface/DeepDoubleCvBTagInfo.h"
+
+#include "RecoBTag/FeatureTools/interface/BoostedDoubleSVTagInfoConverter.h"
+#include "RecoBTag/FeatureTools/interface/ChargedCandidateConverter.h"
+#include "RecoBTag/FeatureTools/interface/JetConverter.h"
+#include "RecoBTag/FeatureTools/interface/SecondaryVertexConverter.h"
+
+#include "RecoBTag/FeatureTools/interface/TrackInfoBuilder.h"
+#include "RecoBTag/FeatureTools/interface/sorting_modules.h"
+
+#include "DataFormats/Candidate/interface/VertexCompositePtrCandidate.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+#include "RecoBTag/FeatureTools/interface/deep_helpers.h"
+
+class DeepDoubleCvBTagInfoProducer : public edm::stream::EDProducer<>
+{
+
+public:
+    explicit DeepDoubleCvBTagInfoProducer(const edm::ParameterSet&);
+    ~DeepDoubleCvBTagInfoProducer() override;
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+    typedef std::vector<reco::DeepDoubleCvBTagInfo> DeepDoubleCvBTagInfoCollection;
+    typedef reco::VertexCompositePtrCandidateCollection SVCollection;
+    typedef reco::VertexCollection VertexCollection;
+    typedef edm::View<reco::BoostedDoubleSVTagInfo>
+        BoostedDoubleSVTagInfoCollection;
+
+    void beginStream(edm::StreamID) override
+    {
+    }
```

Comment:
there is no need to reimplement an empty method.
Only ```produce``` is pure virtual in the base and is required to be implemented